common/src/main/kotlin/entity/Interactions.kt

```diff
@@ -135,37 +144,65 @@ public sealed class ApplicationCommandOptionType(public val type: Int) {
 @Serializable(Choice.Serializer::class)
 public sealed class Choice<out T> {
     public abstract val name: String
+    public abstract val nameLocalizations: Optional<Map<Locale, String>>
     public abstract val value: T
 
-    public data class IntChoice(override val name: String, override val value: Long) : Choice<Long>()
-    public data class NumberChoice(override val name: String, override val value: Double) : Choice<Double>()
-    public data class StringChoice(override val name: String, override val value: String) : Choice<String>()
+    public data class IntChoice(
+        override val name: String,
+        override val nameLocalizations: Optional<Map<Locale, String>>,
+        override val value: Long
+    ) : Choice<Long>()
+
+    public data class NumberChoice(
+        override val name: String,
+        override val nameLocalizations: Optional<Map<Locale, String>>,
+        override val value: Double
+    ) : Choice<Double>()
+
+    public data class StringChoice(
+        override val name: String,
+        override val nameLocalizations: Optional<Map<Locale, String>>,
+        override val value: String
+    ) : Choice<String>()
 
     internal class Serializer<T>(serializer: KSerializer<T>) : KSerializer<Choice<*>> {
+        private val nameLocalizationsSerializer = run {
+            val mapSerializer = MapSerializer(
+                Locale.serializer(),
+                String.serializer()
+            )
+
+            Optional.serializer(mapSerializer)
+        }
```

Comment:
still constructed for every instance for this serializer, could be top-level or in companion

Reply:
Normally serializers are objects, I forgot that this one isn't